### PR TITLE
docs: golang SDK v0.2.1 release docs

### PR DESCRIPTION
**Pull request by suluvai**
## Description

Added docs for Golang SDK v0.2.1 release.

## Resolved issues
Jira: DX-1274

### Before submitting the PR, please consider the following:
- [x] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems the pull request solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [x] If you are adding or updating a documentation page, please also add or update the team ownership of that page (follow [this guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1947927045/Docs+ownership+registry#What-to-do-when-adding%2Fupdating-documentation-pages%3F)).
- [x] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [X] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

File: sdks-core-sdk-golang_versioned_docs/version-0.2.0/additional-info.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+### Smart contract autogeneration
+
+The Immutable Solidity contracts can be found in the `contracts` folder. Contract bindings in Golang are generated using [abigen](https://geth.ethereum.org/docs/dapp/native-bindings#abigen-go-binding-generator).
+
+#### Core
+
+The Core contract is Immutable's main interface with the Ethereum blockchain, based on [StarkEx](https://docs.starkware.co/starkex-v4).
+
+[View contract](./solidity/Core.sol)
```
Comment (zatlite) [inline sdks-core-sdk-golang_versioned_docs/version-0.2.0/additional-info.md:215]: Broken link

File: sdks-core-sdk-golang_versioned_docs/version-0.2.0/additional-info.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+#### Registration
+
+The Registration contract is a proxy smart contract for the Core contract that combines transactions related to onchain registration, deposits and withdrawals. When a user who is not registered onchain attempts to perform a deposit or a withdrawal, the Registration combines requests to the Core contract in order to register the user first. Users who are not registered onchain are not able to perform a deposit or withdrawal.
+
+For example, instead of making subsequent transaction requests to the Core contract, i.e. `registerUser` and `depositNft`, a single transaction request can be made to the proxy Registration contract - `registerAndWithdrawNft`.
+
+[View contract](./solidity/Registration.sol)
```
Comment (zatlite) [inline sdks-core-sdk-golang_versioned_docs/version-0.2.0/additional-info.md:223]: Broken link here too

File: sdks-core-sdk-golang_versioned_sidebars/version-0.2.0-sidebars.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "additional-info",
+        {
+          "type": "link",
+          "label": "Examples",
+          "href": "https://github.com/immutable/imx-core-sdk-golang/tree/v0.2.0/imx/examples"
+        },
+        {
+          "type": "link",
+          "label": "API client reference",
+          "href": "https://docs.x.immutable.com/sdk-references/core-sdk-golang/0-2-0/apiclient/"
```
Comment (zatlite) [inline sdks-core-sdk-golang_versioned_sidebars/version-0.2.0-sidebars.json:19]: This link is broken.

Comment (zatlite) [reply sdks-core-sdk-golang_versioned_sidebars/version-0.2.0-sidebars.json:19]: Please check that the links inside apiclient reference are also routing correctly.

File: sdks-core-sdk-golang_versioned_docs/version-0.2.0/additional-info.md (hunk 2 above)
Comment (suluvai) [reply sdks-core-sdk-golang_versioned_docs/version-0.2.0/additional-info.md:223]: Was not picked up during the readme

**Review by zatlite — APPROVED** [review]

**Review by Kastet — APPROVED** [review]

**Review by ckailash — APPROVED** [review]

**Pull request merged**